run_mount.go

```diff
@@ -69,12 +77,59 @@ func generateMountOpts(clicontext *cli.Context, imageVolumes map[string]struct{}
 			continue
 		}
 		anonVolName := idgen.GenerateID()
+
 		logrus.Debugf("creating anonymous volume %q, for \"VOLUME %s\"",
 			anonVolName, imgVolRaw)
 		anonVol, err := volStore.Create(anonVolName)
 		if err != nil {
 			return nil, nil, err
 		}
+
+		client, ctx, cancel, err := newClient(clicontext)
+		if err != nil {
+			return nil, nil, err
+		}
+		defer cancel()
+
+		img, err := client.ImageService().Get(ctx, clicontext.Args().First())
+		if err != nil {
+			return nil, nil, err
+		}
+
+		i := containerd.NewImage(client, img)
+
+		if err := i.Unpack(ctx, clicontext.String("snapshotter")); err != nil {
+			return nil, nil, errors.Wrap(err, "error unpacking image")
+		}
+
+		diffIDs, err := i.RootFS(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
+		chainID := identity.ChainID(diffIDs).String()
+
+		s := client.SnapshotService(clicontext.String("snapshotter"))
+
+		tempDir, err := ioutil.TempDir(os.Getenv("GOTMPDIR"), "initialC")
```

Comment:
Why GOTMPDIR

Reply:
it is juste a choice. I can create tempDir in `.` it doesn't change anything for me

Reply:
updated to `tempDir, err := ioutil.TempDir("", "initialC")`